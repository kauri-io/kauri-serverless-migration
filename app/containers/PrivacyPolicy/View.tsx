import React, { Component } from 'react'
import styled from 'styled-components'

const Content = styled.section`
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 85vh;
    width: 100%;
`

interface IProps {
    routeChangeAction: (route: string) => void
}

type State = {}

class PrivacyPolicy extends Component<IProps, State> {
    render() {
        return (
            <section>
                <Content>
                    <iframe
                        style={{
                            minHeight: '85vh',
                            width: '60%',
                            border: 'none',
                        }}
                        src="https://docs.google.com/document/d/e/2PACX-1vQL9NX53wUMwWgiWQJnSuGne-gmKWb_dhauDt1au9rwsZfjnzeYFuO1ftfyZ28wSCGj10f-tWo3HJJ6/pub?embedded=true"
                    />
                </Content>
            </section>
        )
    }
}

export default PrivacyPolicy
