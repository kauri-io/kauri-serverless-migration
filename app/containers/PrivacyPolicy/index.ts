import { compose } from 'react-apollo'
import { connect } from 'react-redux'
import withLoading from '../../lib/with-loading'
import { routeChangeAction } from '../../lib/Epics/RouteChangeEpic'
import PrivacyPolicy from './View'

export default compose(
    connect(
        () => ({}),
        { routeChangeAction }
    ),
    withLoading()
)(PrivacyPolicy)
