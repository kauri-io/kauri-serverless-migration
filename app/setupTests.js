import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
