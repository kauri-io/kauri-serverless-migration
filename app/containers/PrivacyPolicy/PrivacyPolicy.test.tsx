import PrivacyPolicy from './index'
import { mountWithRedux } from '../../setupTests'

describe('containers/PrivacyPolicy', () => {
    it('snapshot should match', () => {
        const props = {}
        const wrapper = mountWithRedux(<PrivacyPolicy {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
