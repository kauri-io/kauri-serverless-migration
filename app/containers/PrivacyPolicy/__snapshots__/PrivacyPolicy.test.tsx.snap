// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/PrivacyPolicy snapshot should match 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <Connect(branch(PrivacyPolicy))>
      <branch(PrivacyPolicy)
        routeChangeAction={[Function]}
      >
        <PrivacyPolicy
          routeChangeAction={[Function]}
        >
          <section>
            <View__Content>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "View__Content-sc-1jhb3py-0",
                      "isStatic": true,
                      "lastClassName": "gGsUfO",
                      "rules": Array [
                        "display:flex;justify-content:center;align-items:center;min-height:85vh;width:100%;",
                      ],
                    },
                    "displayName": "View__Content",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "View__Content-sc-1jhb3py-0",
                    "target": "section",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <section
                  className="View__Content-sc-1jhb3py-0 gGsUfO"
                >
                  <iframe
                    src="https://docs.google.com/document/d/e/2PACX-1vQL9NX53wUMwWgiWQJnSuGne-gmKWb_dhauDt1au9rwsZfjnzeYFuO1ftfyZ28wSCGj10f-tWo3HJJ6/pub?embedded=true"
                    style={
                      Object {
                        "border": "none",
                        "minHeight": "85vh",
                        "width": "60%",
                      }
                    }
                  />
                </section>
              </StyledComponent>
            </View__Content>
          </section>
        </PrivacyPolicy>
      </branch(PrivacyPolicy)>
    </Connect(branch(PrivacyPolicy))>
  </ThemeProvider>
</Provider>
`;
